pygama/dsp/_processors/multi_t_filter.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+from numba import guvectorize, njit
+from time_point_thresh import time_point_thresh
+from get_multi_local_extrema import get_multi_local_extrema
+
+
+    
+@guvectorize(["void(float32[:],float32[:])",
+              "void(float64[:],float64[:])",
+              "void(int32[:],int32[:])",
+              "void(int64[:],int64[:])"],
+             "(m),(m)", nopython=True, cache=True)
+def remove_duplicates(t_out,vt_min_out):
+    """ 
+    time_point_thresh has issues with afterpulsing in waveforms that causes  
+    an aferpulse peak's tp0 to be sent to 0 or the same index as the tp0 for the first pulse.
+    This only happens when the relative minimum between the first pulse and 
+    the afterpulse is greater than the threshold. So, we sweep through the array again 
+    to ensure there are no duplicate indices. If there are duplicate indicies caused by a
+    misidentified tp0 of an afterpulse, we replace its index by that of the corresponding minimum
+    found using the njit_get_i_local_extrema function.
+    ----------
+    t_out : array-like
+        the array of indices that we want to remove duplicates from 
+    vt_min_out : array-like
+        list of indicies of minima that we want to replace duplicates in t_out with
+    """
+
```

Comment:
add initializations and checks, see points 5 and 6: https://github.com/legend-exp/pygama/blob/5e261e69eec1610431fe2eda42d994ba58d299ea/pygama/dsp/_processors/template.py#L39